src/components/molecules/DataWrapper.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { ReactNode, useEffect, useState } from 'react';
+import { ApolloError, DocumentNode, useQuery } from '@apollo/client';
+import { TimeEntry } from '../atoms/ThresholdChart';
+import { formatMetData } from '../../utils/formatMetData';
+
+export type DataWrapperProps = {
+    query: DocumentNode;
+    children: (args: { data: TimeEntry[] | undefined; loading: boolean; error: ApolloError | undefined }) => ReactNode;
+};
+
+const DataWrapper: React.FC<DataWrapperProps> = ({ children, query }) => {
+    const [formattedData, setFormattedData] = useState<TimeEntry[] | undefined>(undefined);
```

Comment:
```suggestion
    const [formattedData, setFormattedData] = useState<TimeEntry[]>([]);
```

Reply:
Default to empty formattedData instead might be easier to type. Just a suggestion.

If not, I believe you can omit `| undefinned` and the `undefined` in the argument list, and it will default to `null`, which I believe is more semantically correct here.